framework/include/base/SubProblem.h

```diff
@@ -489,6 +500,9 @@ class SubProblem : public Problem
   std::string restrictionCheckName(BoundaryID check_id);
   ///@}
 
+  /// Flag to determine whether the problem is currently computing Jacobian
+  bool _currently_computing_jacobian;
```

Comment:
There's already a member with this same name in FEProblemBase, which derives from this, so this makes that one a shadow variable, which is a problem. It looks like the only reason you're doing any of this is because you have a SubProblem in the geometric search code rather than an FEProblem in NearestNodeLocator. Maybe it's better to switch the SubProblem object that GeometricSearchData and these other related objects are passing around to be an FEProblemBase (look at the constructor to GeometricSearchInterface, which is where we're getting that). Either that or dynamic cast _subproblem to an FEProblemBase.

I don't really know too much about the thought behind the design of the hierarchy of SubProblem->FEProblemBase->FEProblem, so I don't know if there's really any good reason that we're interfacing with SubProblem in this geometric search stuff rather than FEProblemBase. @permcody can you shed any light on that?

Reply:
The original design was that we'd have different types of Problems. FEProblem being the "normal" MOOSE solve (one NL system, and one Aux system) but originally we didn't want to restrict people to this limitation. It has in-fact been utilized with the EigenProblem generalization. However, we ran with the single FEProblem instance for so long that it eventually creeped it's way into several systems. Some systems have the parent SubProblem class reference while others have actual FEProblem references. The latter can't be easily used with EigenProblems though which is what we wanted to avoid.

This Boolean should probably be removed from FEProblem if it's going to be moved to here. I think we should avoid the shadowing.

Reply:
We definitely don't want shadowing. We really have two options here:
1) Move this Boolean to Subproblem
2) Change the geometric search system to work with a reference to FEProblemBase rather than SubProblem.
If we go with option 1, does that pollute SubProblem with something that is too specific to FEProblem? I'm guessing maybe not -- the EigenProblem probably has something similar. It sounds like the drawback to option 2 is that then the geometric search system wouldn't work with EigenProblem. Sounds like maybe option 1 is the best.

Reply:
Thanks for the comments @bwspenc  and @permcody. 

So, it looks like moving the Boolean from FEProblem to SubProblem is the better option. I will do that and see if it breaks anything.

Reply:
I have moved the boolean `_currently_computing_jacobian` from FEProblem to SubProblem.